File: src/main/java/com/range/rpms/exception/GlobalExceptionHandler.java
package com.range.rpms.exception;

import com.range.rpms.dto.GenericResponse;
import com.range.rpms.util.ErrorResponseUtil;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

@ControllerAdvice
public class GlobalExceptionHandler {
    @ExceptionHandler(AbstractExceptionHandler.class)
    public ResponseEntity<GenericResponse<Void>> handleException(AbstractExceptionHandler e) {

        return ErrorResponseUtil
                .build(e.getHttpStatus()
                        ,e.getMessage()
                );

    }

}

File: src/main/kotlin/com/example/rpms/common/exception/GlobalExceptionHandler.kt
package com.example.rpms.common.exception

import com.example.rpms.common.dto.ExceptionResponse
import jakarta.servlet.http.HttpServletRequest
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler

@ControllerAdvice
class GlobalExceptionHandler
{
    @ExceptionHandler(BaseException::class)
    fun handleBaseException(ex:BaseException,
                            request: HttpServletRequest
    ): ExceptionResponse{
        return ExceptionResponse(ex.localizedMessage, request.requestURI,ex.status)
    }
}